src/persist-client/src/internal/state.proto

```diff
@@ -41,19 +41,25 @@ message ProtoTrace {
 }
 
 message ProtoLeasedReaderState {
+    string hostname = 5;
+    string description = 6;
     ProtoU64Antichain since = 1;
     uint64 seqno = 2;
     uint64 last_heartbeat_timestamp_ms = 3;
     uint64 lease_duration_ms = 4;
 }
 
 message ProtoCriticalReaderState {
+    string hostname = 4;
+    string description = 5;
```

Comment:
is it worth pulling out these two fields into a `ProtoHandle(Debug|Audit|Diagnostic)` message?

Reply:
what's the benefit?

Reply:
the type of information in them feel a bit different than the existing fields. e.g. `since` and `opaque` are semantic values of the reader state, and the `hostname` and `purpose` are more metadata

was also wondering if it might help unify the leased reader/critical reader/writer codepaths elsewhere

Reply:
done. the review commits were harder to reason about than just force pushing, so went ahead and did that and rebased in the conflict